src/pages/Login/api/useLoginQuery.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMutation, useQueryClient } from 'react-query';
+import { loginRequest } from './request/loginRequest';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { useAuth } from '../../../context/authContext';
+
+type LoginData = {
+  email: string;
+  password: string;
+};
+
+type LocationProps = {
+  state: {
+    from: Location;
+  };
+};
+
+const useLoginQuery = () => {
+  const { isUser, isAdmin } = useAuth();
+  const navigate = useNavigate();
+  const location = useLocation() as unknown as LocationProps;
+
+  const from = location.state?.from?.pathname || '/';
+
+  const queryClient = useQueryClient();
+
+  return useMutation((formData: LoginData) => loginRequest(formData), {
```

Comment:
you can name it, e.g.: `loginUser = useMutation(...`

and at the end return it `return { loginUser }`